cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -197,6 +198,38 @@ public void run() {
     return result;
   }
 
+  /**
+   * Demote brokers by removing the leader off the broker.
+   * @param brokerIds the broker ids to move off.
+   * @param dryRun whether it is a dry run or not
+   * @param operationProgress the progress of the job to report.
+   * @return the optimization result.
+   */
+  public GoalOptimizer.OptimizerResult demoteBrokers(Collection<Integer> brokerIds,
+                                                     boolean dryRun,
+                                                     OperationProgress operationProgress)
+      throws KafkaCruiseControlException {
+    PreferredLeaderElectionGoal goal = new PreferredLeaderElectionGoal();
+    try (AutoCloseable ignored = _loadMonitor.acquireForModelGeneration(operationProgress)) {
+      ClusterModel clusterModel = _loadMonitor.clusterModel(_time.milliseconds(),
+                                                            goal.clusterModelCompletenessRequirements(),
+                                                            operationProgress);
+      brokerIds.forEach(id -> clusterModel.setBrokerState(id, Broker.State.DEAD));
```

Comment:
We added a `DEMOTED` state, but here we set the brokers to be demoted as `DEAD`. Our unit test also relies on setting the state to `DEMOTED`. Should we change this line to `State.DEMOTED`?